pkg/cloud/packet/client.go

```diff
@@ -55,17 +55,20 @@ var (
 )
 
 type Client struct {
-	*packngo.Client
+	*metal.APIClient
 }
 
 // NewClient creates a new Client for the given Packet credentials
 func NewClient(packetAPIKey string) *Client {
 	token := strings.TrimSpace(packetAPIKey)
 
 	if token != "" {
-		metalClient := &Client{packngo.NewClientWithAuth(clientName, token, nil)}
-		metalClient.UserAgent = fmt.Sprintf(clientUAFormat, version.Get(), metalClient.UserAgent)
-
+		configuration := metal.NewConfiguration()
+		configuration.Debug = true
```

Comment:
Is the Debug flag intended to come out as a merge prerequisite or do we want to conditionally enable this similar to how we had PACKNGO_DEBUG as an option in packngo based builds?

Reply:
I was intending to take this out before merge, but it's probably preferable to allow setting it via an environment variable.

Reply:
This still needs to be removed/converted to a variable, doesn't it?

Reply:
This is still outstanding?

Reply:
Yep, forgot about this one

Reply:
I added a function to check for the `PACKNGO_DEBUG` env var for backwards compatibility.  Is there a standard variable in k8s/cluster-api/CNCF land that we should look at as well?  (For example, our terraform provider supports turning on debug logs based on the `TF_LOG` environment variable; is there something similar we could do here?)

Reply:
Closing the loop:  we may be able to support other ways of turning on debug logs, but `PACKNGO_DEBUG` is all that's needed for backwards compatibility.  We can follow up separately on alternatives for debug logging.